js-exercises/freq-sort/freqSort.js

```diff
@@ -0,0 +1,25 @@
+function freqSort(arrayOfCharacters) {
+  if (!Array.isArray(arrayOfCharacters)) {
+    throw new Error("argument should be array");
+  } else {
+    let frequencyMap = {};
+
+    arrayOfCharacters.map(checkFreqElement);
+
+    function checkFreqElement(element) {
+      element = element.normalize("NFC");
+      if (frequencyMap.hasOwnProperty(element))
+        return (frequencyMap[element] += 1);
+      else return (frequencyMap[element] = 1);
+    }
+
+    let sortedArray = Object.entries(frequencyMap).sort(
+      (firstElement, secondElement) => secondElement[1] - firstElement[1]
+    );
+
```

Comment:
I think as explained in yesterday's class we can make things declarative here(data transformation style) , making business logic exposed and lang construct hidden